socorro/unittest/signature/test_rules.py

```diff
@@ -1281,58 +1281,73 @@ def test_action_success_long_message(self):
         expected_sig = 'Abort | {}... | hello'.format('a' * 77)
         assert processed_crash['signature'] == expected_sig
 
-    def test_action_success_remove_unwanted_parts(self):
-        rule = AbortSignature()
-
-        raw_crash = {}
-        processed_crash = {}
-
-        # Test with just the "ABOR" thing at the start.
-        processed_crash['signature'] = 'hello'
-        raw_crash['AbortMessage'] = '[5392] ###!!! ABORT: foo bar line 42'
-
-        action_result = rule.action(raw_crash, processed_crash, [])
-
-        assert action_result is True
-        assert processed_crash['original_signature'] == 'hello'
-        assert processed_crash['signature'] == 'Abort | foo bar line 42 | hello'
-
-        # Test with a file name and line number.
-        raw_crash['AbortMessage'] = (
-            '[7616] ###!!! ABORT: unsafe destruction: file '
-            'c:/builds/moz2_slave/m-rel-w32-00000000000000000000/build/src/'
-            'dom/plugins/ipc/PluginModuleParent.cpp, line 777'
-        )
-        processed_crash['signature'] = 'hello'
+    @pytest.mark.parametrize('abort_msg, expected', [
+        # Test with just the "ABOR" thing at the start
+        (
+            '[5392] ###!!! ABORT: foo bar line 42',
+            'Abort | foo bar line 42 | hello'
+        ),
+
+        # Test with a file name and line number
+        (
+            (
+                '[7616] ###!!! ABORT: unsafe destruction: file '
+                'c:/builds/moz2_slave/m-rel-w32-00000000000000000000/build/src/'
+                'dom/plugins/ipc/PluginModuleParent.cpp, line 777'
+            ),
+            'Abort | unsafe destruction | hello'
+        ),
 
-        action_result = rule.action(raw_crash, processed_crash, [])
+        # Test with a message that lacks interesting content.
+        (
+            '[204] ###!!! ABORT: file ?, ',
+            'Abort | hello'
+        ),
 
-        assert action_result is True
-        assert processed_crash['original_signature'] == 'hello'
-        assert processed_crash['signature'] == 'Abort | unsafe destruction | hello'
+        # Test with another message that lacks interesting content.
+        (
+            (
+                '[4648] ###!!! ABORT: file resource:///modules/sessionstore/'
+                'SessionStore.jsm, line 1459'
+            ),
+            'Abort | hello'
+        ),
+
+        # Test with "unable to find a usable font" case
+        (
+            u'unable to find a usable font (\u5fae\u8f6f\u96c5\u9ed1)',
+            'Abort | unable to find a usable font | hello'
+        ),
```

Comment:
This is the new test case that was added.

Reply:
This seems fine. I was worried earlier about destroying information and if it was appropriate to combine these. Having some time to consider it, I now suspect it is a good move and like the new signature.